### PR TITLE
Updating MSBuild reference

### PR DESCRIPTION
**Pull request by v-alje**
Update the MSBuild reference from the dlls and XML in their NuGet packages.

CEAPEX PROJECT 314046 MSBuild 16.7- .NET

### CODE REVIEW

File: xml/Microsoft.Build.Framework/SdkResult.xml [hunk 1]
```diff
@@ -44,6 +49,53 @@
...
+      <AssemblyInfo>
+        <AssemblyName>Microsoft.Build.Framework</AssemblyName>
+        <AssemblyVersion>15.1.0.0</AssemblyVersion>
+      </AssemblyInfo>
+      <ReturnValue>
+        <ReturnType>System.Collections.Generic.IDictionary&lt;System.String,Microsoft.Build.Framework.SdkResultItem&gt;</ReturnType>
+      </ReturnValue>
+      <Docs>
+        <summary>
+            Items that should be added to the evaluation.  This allows an SDK resolver to provide information to the build
```
Comment (gewarren) [inline xml/Microsoft.Build.Framework/SdkResult.xml:93]:
```suggestion
            Gets or sets items that should be added to the evaluation. This allows an SDK resolver to provide information to the build.
```

File: xml/Microsoft.Build.Framework/SdkResult.xml [hunk 2]
```diff
@@ -60,11 +112,38 @@
         <ReturnType>System.String</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets the resolved path to the SDK.</summary>
+        <summary>
+                Resolved path to the SDK.
+            
+                Null if <see cref="P:Microsoft.Build.Framework.SdkResult.Success" /> == false
```
Comment (gewarren) [inline xml/Microsoft.Build.Framework/SdkResult.xml:118]:
```suggestion
                Null if <see cref="P:Microsoft.Build.Framework.SdkResult.Success" /> is <see langword="false" />.
```

File: xml/Microsoft.Build.Framework/SdkResult.xml [hunk 3]
```diff
@@ -60,11 +112,38 @@
...
+      <AssemblyInfo>
+        <AssemblyName>Microsoft.Build.Framework</AssemblyName>
+        <AssemblyVersion>15.1.0.0</AssemblyVersion>
+      </AssemblyInfo>
+      <ReturnValue>
+        <ReturnType>System.Collections.Generic.IDictionary&lt;System.String,System.String&gt;</ReturnType>
+      </ReturnValue>
+      <Docs>
+        <summary>
+            Properties that should be added to the evaluation.  This allows an SDK resolver to provide information to the build
```
Comment (gewarren) [inline xml/Microsoft.Build.Framework/SdkResult.xml:141]:
```suggestion
Gets or sets properties to add to the evaluation. This allows an SDK resolver to provide information to the build.
```

File: xml/Microsoft.Build.Framework/SdkResult.xml [hunk 4]
```diff
@@ -81,7 +160,9 @@
         <ReturnType>Microsoft.Build.Framework.SdkReference</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets the SDK reference.</summary>
+        <summary>
+                The Sdk reference
```
Comment (gewarren) [inline xml/Microsoft.Build.Framework/SdkResult.xml:164]:
```suggestion
                The SDK reference.
```

File: xml/Microsoft.Build.Framework/SdkResult.xml [hunk 5]
```diff
@@ -102,7 +183,9 @@
         <ReturnType>System.Boolean</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets a value that indicates whether the resolution was successful.</summary>
+        <summary>
+                Indicates the resolution was successful.
```
Comment (gewarren) [inline xml/Microsoft.Build.Framework/SdkResult.xml:187]:
```suggestion
                Indicates whether the resolution was successful.
```

File: xml/Microsoft.Build.Framework/SdkResult.xml [hunk 6]
```diff
@@ -124,8 +207,12 @@
         <ReturnType>System.String</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets the resolved version of the SDK.
-                Can be <see langword="null" /> or empty if the resolver did not provide a version, for example, a path-based resolver.</summary>
+        <summary>
+                Resolved version of the SDK.
+                Can be null or empty if the resolver did not provide a version (e.g. a path based resolver)
```
Comment (gewarren) [inline xml/Microsoft.Build.Framework/SdkResult.xml:212]:
```suggestion
                Can be null or empty if the resolver did not provide a version (for example, a path based resolver).
```

File: xml/Microsoft.Build.Framework/SdkResult.xml [hunk 7]
```diff
@@ -124,8 +207,12 @@
...
+            
+                Null if <see cref="P:Microsoft.Build.Framework.SdkResult.Success" /> == false
```
Comment (gewarren) [inline xml/Microsoft.Build.Framework/SdkResult.xml:214]:
```suggestion
                <see langword="null" /> if <see cref="P:Microsoft.Build.Framework.SdkResult.Success" /> is <see langword="false" />.
```

File: xml/Microsoft.Build.Framework/SdkResultFactory.xml [hunk 8]
```diff
@@ -99,5 +99,44 @@
...
+        <param name="warnings">Optional warnings to display during resolution.</param>
+        <summary>
+                Create an <see cref="T:Microsoft.Build.Framework.SdkResolver" /> object indicating success.
+            </summary>
+        <returns />
+        <remarks>
+            This overload allows any number (zero, one, or many) of SDK paths to be returned.  This means a "successful" result
+            may not resolve to any SDKs.  The resolver can also supply properties or items to communicate information to the build.  This
+            can allow resolvers to report SDKs that could not be resolved without hard-failing the evaluation, which can allow other
+            components to take more appropriate action (for example installing optional workloads or downloading NuGet SDKs).
```
Comment (gewarren) [inline xml/Microsoft.Build.Framework/SdkResultFactory.xml:137]:
```suggestion
            components to take more appropriate action (for example, installing optional workloads or downloading NuGet SDKs).
```

File: xml/Microsoft.Build.Framework/SdkResultItem.xml [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+    <AssemblyName>Microsoft.Build.Framework</AssemblyName>
+    <AssemblyVersion>15.1.0.0</AssemblyVersion>
+  </AssemblyInfo>
+  <Base>
+    <BaseTypeName>System.Object</BaseTypeName>
+  </Base>
+  <Interfaces />
+  <Docs>
+    <summary>
+            The value of an item and any associated metadata to be added by an SDK resolver.  See <see cref="P:Microsoft.Build.Framework.SdkResult.ItemsToAdd" /></summary>
```
Comment (gewarren) [inline xml/Microsoft.Build.Framework/SdkResultItem.xml:18]:
```suggestion
            The value of an item and any associated metadata to be added by an SDK resolver. Also see <see cref="P:Microsoft.Build.Framework.SdkResult.ItemsToAdd" />.</summary>
```

File: xml/Microsoft.Build.Framework/TaskFinishedEventArgs.xml [hunk 10]
```diff
@@ -26,7 +26,6 @@
     <summary>
             Arguments for task finished events
             </summary>
-    <remarks>To be added.</remarks>
```
Comment (gewarren) [inline xml/Microsoft.Build.Framework/TaskFinishedEventArgs.xml:29]: Were these removed manually? There's no need to remove them, they won't show up on the published docs.

File: xml/Microsoft.Build.Framework/TaskPropertyInfo.xml [hunk 11]
```diff
@@ -57,6 +57,52 @@
...
+      <AssemblyInfo>
+        <AssemblyName>Microsoft.Build.Framework</AssemblyName>
+        <AssemblyVersion>15.1.0.0</AssemblyVersion>
+      </AssemblyInfo>
+      <ReturnValue>
+        <ReturnType>System.Boolean</ReturnType>
+      </ReturnValue>
+      <Docs>
+        <summary>
+            This task parameter should be logged when LogTaskInputs is set. Defaults to true.
```
Comment (gewarren) [inline xml/Microsoft.Build.Framework/TaskPropertyInfo.xml:77]:
```suggestion
Gets or sets a value that indicates whether LogTaskInputs is set. The default is <see langword="true" />.
```

File: xml/Microsoft.Build.Framework/TaskPropertyInfo.xml [hunk 12]
```diff
@@ -57,6 +57,52 @@
...
+      <AssemblyInfo>
+        <AssemblyName>Microsoft.Build.Framework</AssemblyName>
+        <AssemblyVersion>15.1.0.0</AssemblyVersion>
+      </AssemblyInfo>
+      <ReturnValue>
+        <ReturnType>System.Boolean</ReturnType>
+      </ReturnValue>
+      <Docs>
+        <summary>
+            When this task parameter is an item list, determines whether to log item metadata. Defaults to true.
```
Comment (gewarren) [inline xml/Microsoft.Build.Framework/TaskPropertyInfo.xml:100]:
```suggestion
Gets or sets a value that indicates whether to log item metadata when this task parameter is an item list. The default is <see langword="true" />.
```

File: xml/Microsoft.Build.Globbing/MSBuildGlobWithGaps.xml [hunk 13]
```diff
@@ -19,8 +19,17 @@
   </Interfaces>
   <Docs>
     <summary>
-    A glob with gaps. The gaps are represented as other globs.
-    </summary>
+                A glob with gaps. The gaps are represented as other globs.
```
Comment (gewarren) [inline xml/Microsoft.Build.Globbing/MSBuildGlobWithGaps.xml:22]: This section is messed up: https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.build.globbing.msbuildglobwithgaps?view=msbuild-15&branch=pr-en-us-4939.

Comment (gewarren) [reply xml/Microsoft.Build.Globbing/MSBuildGlobWithGaps.xml:22]:
```suggestion
A glob with gaps. The gaps are represented as other globs.
```

File: xml/Microsoft.Build.Globbing/MSBuildGlobWithGaps.xml [hunk 14]
```diff
@@ -19,8 +19,17 @@
...
+                For example, to express a glob that matches all .cs files except the ones containing "foo" and the ones under bin directories, one can use:
```
Comment (gewarren) [inline xml/Microsoft.Build.Globbing/MSBuildGlobWithGaps.xml:23]:
```suggestion
For example, to express a glob that matches all .cs files except the ones containing "foo" and the ones under bin directories, one can use:
```

File: xml/Microsoft.Build.Tasks/Hash.xml [hunk 15]
```diff
@@ -14,7 +14,12 @@
   </Base>
   <Interfaces />
   <Docs>
-    <summary> Generates a hash of a given group of items. Metadata is not considered in the hash.</summary>
+    <summary>
+            Generates a hash of a given ItemGroup items. Metadata is not considered in the hash.
+            <remarks>
```
Comment (gewarren) [inline xml/Microsoft.Build.Tasks/Hash.xml:19]:
```suggestion
```

Remarks shouldn't be nested in summary tags.

File: xml/Microsoft.Build.Tasks/Hash.xml [hunk 16]
```diff
@@ -14,7 +14,12 @@
...
+            Currently uses SHA1. Implementation subject to change between MSBuild versions. Not
+            intended as a cryptographic security measure, only uniqueness between build executions.
+            </remarks></summary>
```
Comment (gewarren) [inline xml/Microsoft.Build.Tasks/Hash.xml:22]:
```suggestion
            </summary>
```

File: xml/Microsoft.Build.Tasks/Hash.xml (hunk 15 above)
Comment (gewarren) [reply xml/Microsoft.Build.Tasks/Hash.xml:19]: Here's the preview: https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.build.tasks.hash?view=msbuild-15&branch=pr-en-us-4939.

The remarks should ideally be moved to the Remarks section.

File: xml/Microsoft.Build.Tasks/Hash.xml [hunk 17]
```diff
@@ -62,7 +67,9 @@ This class is not intended as a cryptographic security measure, only for uniquen
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>Executes the task.</summary>
+        <summary>
+            Execute the task.
```
Comment (gewarren) [inline xml/Microsoft.Build.Tasks/Hash.xml:71]:
```suggestion
            Executes the task.
```

File: xml/Microsoft.Build.Tasks/Hash.xml [hunk 18]
```diff
@@ -89,7 +96,9 @@ This class is not intended as a cryptographic security measure, only for uniquen
         <ReturnType>System.String</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets or sets the hash of the <see langword="ItemsToHash ItemSpec" />.</summary>
+        <summary>
+            Hash of the ItemsToHash ItemSpec.
```
Comment (gewarren) [inline xml/Microsoft.Build.Tasks/Hash.xml:100]:
```suggestion
Gets or sets the hash of the <see langword="ItemsToHash ItemSpec" />.
```

File: xml/Microsoft.Build.Tasks/Hash.xml [hunk 19]
```diff
@@ -110,7 +119,9 @@ This class is not intended as a cryptographic security measure, only for uniquen
         <ReturnType>System.Boolean</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets or sets a value that indicates whether to generate a case-insensitive hash.</summary>
+        <summary>
+            When true, will generate a case-insensitive hash.
```
Comment (gewarren) [inline xml/Microsoft.Build.Tasks/Hash.xml:123]:
```suggestion
            Gets or sets a value that indicates whether to generate a case-insensitive hash.
```

File: xml/Microsoft.Build.Tasks/Hash.xml [hunk 20]
```diff
@@ -138,7 +149,9 @@ This class is not intended as a cryptographic security measure, only for uniquen
         <ReturnType>Microsoft.Build.Framework.ITaskItem[]</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets or sets the items from which to generate a hash.</summary>
+        <summary>
+            Items from which to generate a hash.
```
Comment (gewarren) [inline xml/Microsoft.Build.Tasks/Hash.xml:153]:
```suggestion
            Gets or sets the items from which to generate a hash.
```

File: xml/Microsoft.Build.Tasks/ToolTaskExtension.xml [hunk 21]
```diff
@@ -74,6 +74,7 @@
             Fills the provided CommandLineBuilderExtension with those switches and other information that can't go into a response file and
             must go directly onto the command line.
             </summary>
+        <returns>true, if successful</returns>
```
Comment (gewarren) [inline xml/Microsoft.Build.Tasks/ToolTaskExtension.xml:77]:
```suggestion
        <returns><see langword="true" /> if successful.</returns>
```

File: xml/Microsoft.Build.Utilities/Task.xml [hunk 22]
```diff
@@ -369,6 +369,29 @@
...
+      <AssemblyInfo>
+        <AssemblyName>Microsoft.Build.Utilities.Core</AssemblyName>
+        <AssemblyVersion>15.1.0.0</AssemblyVersion>
+      </AssemblyInfo>
+      <ReturnValue>
+        <ReturnType>Microsoft.Build.Framework.IBuildEngine7</ReturnType>
+      </ReturnValue>
+      <Docs>
+        <summary>
+            Retrieves the <see cref="T:Microsoft.Build.Framework.IBuildEngine7" /> version of the build engine interface provided by the host.
```
Comment (gewarren) [inline xml/Microsoft.Build.Utilities/Task.xml:389]:
```suggestion
            Gets the version of the build engine interface provided by the host.
```

The type is shown automatically and should be omitted from the description.

File: xml/Microsoft.Build.Framework/SdkResolverContext.xml [hunk 23]
```diff
@@ -93,7 +99,12 @@
         <ReturnType>System.Version</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets the version of MSBuild that's currently running.</summary>
+        <summary>
+                Version of MSBuild currently running.
+            <remarks>
+                File version based on commit height from our public git repository. This is informational
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply xml/Microsoft.Build.Framework/SdkResolverContext.xml:105]: Also, what is "commit height"?

File: xml/Microsoft.Build.Utilities/MuxLogger.xml [hunk 24]
```diff
@@ -60,7 +60,10 @@
                 At this point we will look up the success state of the project finished event for the submission ID and log a build finished event to the logger.
                 The event source will be cleaned up.  This may be interesting because the unregister will come from a thread other than what is doing the logging.
                 This may create a Synchronization issue, if unregister is called while events are being logged.
-      </summary>
+                
+            UNDONE: If we can use ErrorUtilities, replace all InvalidOperation and Argument exceptions with the appropriate calls.
```
Comment (BillWagner) [inline xml/Microsoft.Build.Utilities/MuxLogger.xml:64]: Do we want to publish these comments? Or were those only for internal use?

Comment (BillWagner) [reply xml/Microsoft.Build.Utilities/MuxLogger.xml:64]: There a a few instances of these.

**Comment by v-alje** [comment]
@ghogen Can you take a look at the review change requests?
Thanks!

**Comment by ghogen** [comment]
@v-alje I don't have permissions to accept the suggested changes.  They all look fine, so you can accept them.  For the remaining, I will need to do the work and submit to the release-msbuild branch.  To clean this up to the standard of this review, we really should also go back to all the other MSBuild comments and search for similar constructs.  But, that could be done separately.

File: xml/Microsoft.Build.Framework/SdkResolverContext.xml (hunk 23 above)
[the remark this answers is not in the document]
Comment (ghogen) [reply xml/Microsoft.Build.Framework/SdkResolverContext.xml:105]: @rainersigwald Should we reference the msbuild GitHub repo in this doc?  https://github.com/Microsoft/msbuild 
Is "commit height" a jargon term?  What can we say instead?

File: xml/Microsoft.Build.Framework/LoggerVerbosity.xml [hunk 25]
```diff
@@ -37,7 +37,10 @@
             4) Detailed -- display all errors, warnings, high and normal importance events, all status events, and a build summary
             5) Diagnostic -- display all events, and a build summary
             
-      </remarks>
+            WARNING: VS Automation code for the Tools/Options MSBuild build verbosity setting will be broken
```
[the remark this answers is not in the document]
Comment (ghogen) [reply xml/Microsoft.Build.Framework/LoggerVerbosity.xml:40]: No, I apparently missed this one in my pass.

File: xml/Microsoft.Build.Execution/ProjectItemGroupTaskItemInstance.xml [hunk 26]
```diff
@@ -408,6 +408,122 @@
...
+          <AttributeName Language="C#">[get: System.Diagnostics.DebuggerStepThrough]</AttributeName>
+          <AttributeName Language="F#">[&lt;get: System.Diagnostics.DebuggerStepThrough&gt;]</AttributeName>
+        </Attribute>
+      </Attributes>
+      <ReturnValue>
+        <ReturnType>System.String</ReturnType>
+      </ReturnValue>
+      <Docs>
+        <summary>
+            Unevaluated MatchOnMetadata value
```
[the remark this answers is not in the document]
Comment (ghogen) [reply xml/Microsoft.Build.Execution/ProjectItemGroupTaskItemInstance.xml]: I think it should be 
Gets the unevaluated <see langword="MatchOnMetadata"> value.

**Comment by gewarren** [comment]
@v-alje On the "Files changed" tab, you batch the suggestions into a single commit so it doesn't kick off a new build for each suggestion that's committed.

File: xml/Microsoft.Build.Execution/ProjectItemGroupTaskItemInstance.xml (hunk 26 above)
[the remark this answers is not in the document]
Comment (gewarren) [reply xml/Microsoft.Build.Execution/ProjectItemGroupTaskItemInstance.xml]: @ghogen Can you add the closing / character in your suggestion?

Gets the unevaluated <see langword="MatchOnMetadata" /> value.

**Comment by v-alje** [comment]
Thank you @gewarren!

File: xml/Microsoft.Build.Utilities/MuxLogger.xml (hunk 24 above)
Comment (ghogen) [reply xml/Microsoft.Build.Utilities/MuxLogger.xml:64]: Removing all instances of UNDONE comments. Thanks.

File: xml/Microsoft.Build.Execution/ProjectInstance.xml [hunk 27]
```diff
@@ -126,6 +127,7 @@
             This is ideal if the project is simply going to be built, and not displayed or edited.
             Uses the default project collection.
             </summary>
+        <returns>A new project instance</returns>
```
Comment (ghogen) [inline xml/Microsoft.Build.Execution/ProjectInstance.xml:130]:
```suggestion
        <returns>A new project instance.</returns>
```

File: xml/Microsoft.Build.Execution/ProjectInstance.xml [hunk 28]
```diff
@@ -164,6 +166,7 @@
...
```
Comment (ghogen) [inline xml/Microsoft.Build.Execution/ProjectInstance.xml:169]:
```suggestion
        <returns>A new project instance.</returns>
```

File: xml/Microsoft.Build.Framework/SdkResolverContext.xml (hunk 23 above)
[the remark this answers is not in the document]
Comment (ghogen) [reply xml/Microsoft.Build.Framework/SdkResolverContext.xml:105]: I rewrote this and submitted the change to the release-msbuild branch.

**Comment by ghogen** [comment]
Some additional fixes in https://github.com/dotnet/dotnet-api-docs/pull/4994
Not sure what to do about CircularDependencyException

**Comment by rainersigwald** [comment]
We should make that type public, but removing it is totally fine for this version. https://github.com/dotnet/msbuild/issues/5811

**Review by ghogen — APPROVED** [review]

**Review by BillWagner — APPROVED** [review]
This is ready to merge.

Thanks @v-alje and @ghogen

**Pull request merged**